### ISSUE TITLE
Create an overlay system for controllers

### ISSUE THREAD

**Issue by digitalLumberjack** [recalbox/recalbox-os#554]
Today the only way to customize controls for a system is to override the whole configuration file.

The best feature i can imagine is to create overlays that will override the configuration of the controller with mappings like : 
overlay {
  a -> b
  b -> a
}

This overlay should be read at ES startup and could be attributed to a specific game, saved in its metadata.

And then, we won. We can have a arcade6btn.overlay for arcade 6btn games, an snk-to-front-buttons.overlay for having snk buttons on A B X Y etc etc.

**Comment by julianxhokaxhiu** [comment]
+1

**Comment by substring** [comment]
yeah, but that may not be that simple. The syntax for controllers depends on the emulator. It may work fine with retroarch, but the mapping options differ in mupen, moonlight, maybe even kodi :
  - file format
  - buttons names
  - the right string to get from the input controller (need the id ? the value ? the code ? how to manage reverted axis ?)

**Comment by digitalLumberjack** [comment]
You look the same you two ...
Subs : you are misunderstanding what i want to do, it would just be a mapping that would switch a <-> b for example, so then you just use configgen as if you received informations from the input_cfg.
